core/Types.h

```diff
@@ -219,7 +219,7 @@ class Types final {
     // something that look like type syntax in a method body.
     static TypePtr applyTypeArguments(const GlobalState &gs, const CallLocs &locs, uint16_t numPosArgs,
                                       const InlinedVector<const TypeAndOrigins *, 2> &args,
-                                      ClassOrModuleRef genericClass);
+                                      ClassOrModuleRef genericClass, bool inResolver);
```

Comment:
Where did we stand on passing in the error class to use to report the error, instead of having a boolean to figure that out inside `core/`?

In some sense, doing it this way is a layering violation: `core/` shouldn't have knowledge of the pipeline, it should be helpers that only really deal with types and classes and symbols that apply no matter the phase of Sorbet.

Reply:
Sorry, I don't remember that conversation. 
I have no issue with passing in the error class instead of a boolean. It works for me!